IDE/Espressif/ESP-IDF/examples/wolfssl_test/main/main.c

```diff
@@ -126,6 +128,44 @@ void my_atmel_free(int slotId)
 void app_main(void)
 {
     int rc = 0;
+    ESP_LOGI(TAG, "--------------------------------------------------------");
+    ESP_LOGI(TAG, "--------------------------------------------------------");
+    ESP_LOGI(TAG, "---------------------- BEGIN MAIN ----------------------");
+    ESP_LOGI(TAG, "--------------------------------------------------------");
+    ESP_LOGI(TAG, "--------------------------------------------------------");
+
+
+    ESP_LOGI(TAG, "CONFIG_IDF_TARGET = %s", CONFIG_IDF_TARGET);
+    ESP_LOGI(TAG, "LIBWOLFSSL_VERSION_STRING = %s", LIBWOLFSSL_VERSION_STRING);
+
+#if defined(LIBWOLFSSL_VERSION_GIT_HASH)
+    ESP_LOGI(TAG, "LIBWOLFSSL_VERSION_GIT_HASH = %s", LIBWOLFSSL_VERSION_GIT_HASH);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_GIT_SHORT_HASH )
+    ESP_LOGI(TAG, "LIBWOLFSSL_VERSION_GIT_SHORT_HASH = %s", LIBWOLFSSL_VERSION_GIT_SHORT_HASH);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_GIT_HASH_DATE)
+    ESP_LOGI(TAG, "LIBWOLFSSL_VERSION_GIT_HASH_DATE = %s", LIBWOLFSSL_VERSION_GIT_HASH_DATE);
+#endif
+
+
+#if defined(CONFIG_IDF_TARGET_ESP32C3)
+    /* not available for C3 at this time */
+#else
+   // ESP_LOGI(TAG, "CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ = %u MHz", CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ);
```

Comment:
Please cleanup this line.

Reply:
As yes, good. Thanks for finding that. That was not intended.

I've added something a bit more clear and graceful, although not yet the final product to fully and properly support all of the chipsets. This should bring operational support without HW acceleration for the ESP32-S3 (cc: @PaulMartinsen)

btw - note there's a bit of chicken-and-egg issue with the wolfSSL `user_settings.h`  and target Espressif chip selection at this time. My user settings has this logic at the tail of of my file to detect the Espressif target from the `sdkconfig.h` file and _conditionally_ turn off HW acceleration:

```
#include "sdkconfig.h"

/* adjust wait-timeout count if you see timeout in rsa hw acceleration */
#define ESP_RSA_TIMEOUT_CNT    0x249F00
#if defined(CONFIG_IDF_TARGET_ESP32C3)
    #define NO_ESP32WROOM32_CRYPT
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_AES
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI
#elif defined(CONFIG_IDF_TARGET_ESP32S3)
    #define NO_ESP32WROOM32_CRYPT
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_AES
    #define NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI
#elif defined(CONFIG_IDF_TARGET_ESP_OTHER)
   /* other */
#else
  /* default is ESP32-WROOM */
#endif
```

One can of course simply ensure hardware acceleration is _always_ off in the `user_settings.h` without detecting architecture. 

Without turning off hardware acceleration for things like the `ESP32-S3` at this time, there will be unexpected and unintuitive compile-time errors.

Also note during this update I saw what appeared to be runtime memory corruption not detected at flash time when doing this:

```
idf.py  -b 921600  -p /dev/ttyS8  build flash monitor
```

I fixed it by doing a `idf.py fullclean`, deleting the existing `sdkconfig` and then loading at a slower [baud rate](https://docs.espressif.com/projects/esptool/en/latest/esp32/esptool/basic-options.html#baud-rate):

```
idf.py  -b 230400  -p /dev/ttyS8 set-target esp32-s3 build flash monitor
```

And for the regular ESP32-WROOM32:

```
idf.py  -b 230400  -p /dev/ttyS15 set-target esp32 build flash monitor
```

Also - I added the known-good `sdkconfig.defaults` currently used in my [WIP Espressif multichip branch](https://github.com/gojimmypi/wolfssl/tree/Espressif_Multichip/IDE/Espressif/ESP-IDF/examples/wolfssl_test).

The ESP32 `wolfssl_test` was confirmed to work with both the ESP32 and ESP32-S3.